sdk/eventhub/event-hubs/test/receiver.spec.ts

```diff
@@ -622,327 +621,6 @@ describe("EventHub Receiver #RunnableInBrowser", function(): void {
-      void
-    > {
-      const partitionId = partitionIds[0];
-      const time = Date.now();
-      // send a message that can be received
-      const sender = client.createProducer({ partitionId });
-      try {
-        await sender.send({ body: "getEventIterator post-cancellation - timeout 0" });
-      } finally {
-        await sender.close();
-      }
-
-      receiver = client.createConsumer(
-        EventHubClient.defaultConsumerGroupName,
-        partitionId,
-        EventPosition.fromEnqueuedTime(time)
-      );
-
-      // abortSignal event listeners will be triggered after synchronous paths are executed
-      const abortSignal = AbortController.timeout(0);
-      const eventIterator = receiver.getEventIterator({ abortSignal });
-      try {
-        for await (const _ of eventIterator) {
-        }
-        throw new Error(`Test failure`);
-      } catch (err) {
-        err.name.should.equal("AbortError");
-        err.message.should.equal("The receive operation has been cancelled by the user.");
-
-        const eventIterator2 = receiver.getEventIterator();
-        try {
-          for await (const _ of eventIterator2) {
-          }
-          throw new Error(`Test failure`);
-        } catch (err) {
-          err.message.should.match(
-            /The EventHubConsumer for ".+" has been closed and can no longer be used.*/gi
-          );
-        }
-      }
-    });
-  });
-
-  describe("mix and match receive methods", function(): void {
-    it("should maintain sequence across calls", function(done: Mocha.Done): void {
-      // wrap test to allow mixing async/await with done
-      (async (done: Mocha.Done) => {
-        const partitionId = partitionIds[0];
-
-        const time = Date.now();
-        // ensure messages exist to retrieve
-        const messageCount = 25;
-        const sentEventData: EventData[] = [];
-        for (let i = 0; i < messageCount; i++) {
-          sentEventData.push({
-            body: `mix and match test messsage ${Date.now()} - ${i}`
-          });
-        }
-
-        const sender = client.createProducer({ partitionId: partitionId });
-        try {
-          await sender.send(sentEventData);
-        } finally {
-          await sender.close();
-        }
-
-        const data: ReceivedEventData[] = [];
-        receiver = client.createConsumer(
-          EventHubClient.defaultConsumerGroupName,
-          partitionId,
-          EventPosition.fromEnqueuedTime(time)
-        );
-
-        // start with iterator
-        for await (const event of receiver.getEventIterator()) {
-          data.push(event);
-          if (data.length >= 5) {
-            break;
-          }
-        }
-
-        // switch to batcher
-        (await receiver.receiveBatch(5)).forEach((event) => {
-          data.push(event);
-        });
-
-        // switch to handler
-        let handlerReceivedCount = 0;
-        const handler = receiver.receive(
```

Comment:
We are planning to have both `receive` and the `receiveBatch`, though only one of them is getting used today?

Reply:
If yes, then we should have a variation of this test that will ensure that `receive` can pick up from where `receiveBatch` left off and vice versa

Reply:
Let me see if that's covered in another test. If not I'll partly resurrect this one. :)

Reply:
cc @chradek as he had added the initial set here

Reply:
cc @chradek as he had added the initial set here

Reply:
cc @chradek as he had added the initial set here

Reply:
I don’t think there’s a need for the firehose receive method anymore. We use receiveBatch behind the scenes for event processor so that we can support back-pressure when receiving events.

It’s also much easier now to support a prefetch queue when using receiveBatch due to the refactoring we did to the underlying receiver this summer, so I don’t think we’ll need to return to receive someday to get better performance.